File: preview/understand/task/create-quickstart/create-quickstart.5.x.cs
// Install the C# / .NET helper library from twilio.com/docs/csharp/install

using System;
using System.Collections.Generic;
using Twilio;
using Twilio.Rest.Preview.Understand.Assistant;


class Program 
{
    static void Main(string[] args)
    {
        // Find your Account Sid and Token at twilio.com/console
        const string accountSid = "ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
        const string authToken = "your_auth_token";

        TwilioClient.Init(accountSid, authToken);

        var actions = new Dictionary<string, Object>()
        {
            {"actions", new object [] {
                new Dictionary<string, Object>()
                {
                    {"say", new Dictionary<string, Object>()
                        {
                            {"speech", "I was going to look for my missing watch, but I could never find the time."}
                        }}
                }
            }}
        };

        var task = TaskResource.Create(
            friendlyName: "tell a joke",
            actions: actions,
            uniqueName: "tell-a-joke",
            pathAssistantSid: "UAXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
        );

        Console.WriteLine(task.Sid);
    }
}

File: preview/understand/task/create-quickstart/create-quickstart.5.x.rb
# Download the helper library from https://www.twilio.com/docs/ruby/install
require 'rubygems'
require 'twilio-ruby'

# Your Account Sid and Auth Token from twilio.com/console
account_sid = 'ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
auth_token = 'your_auth_token'
@client = Twilio::REST::Client.new(account_sid, auth_token)

task = @client.preview.understand
  .assistants('UAXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX')
  .tasks
  .create(friendly_name: 'tell a joke', actions: {
       'actions' => [
           {
               'say' => {
                   'speech' => 'I was going to look for my missing watch, but I could never find the time.'
               }
           },
       ]
   }, unique_name: 'tell-a-joke')

puts task.sid
